code/modules/mob/living/carbon/xenomorph/castes/pyrogen/abilities_pyrogen.dm

```diff
@@ -198,9 +197,10 @@
 			qdel(src)
 			return
 	else if(isobj(target) && !istype(target, /obj/effect/xenomorph/firenado))
-		var/obj/object = target
-		object.take_damage(PYROGEN_TORNADE_HIT_DAMAGE, BURN)
-		qdel(src)
+		if(!istype(target, /obj/structure/mineral_door/resin))
```

Comment:
smelly istype check. This presumably happens with all xeno structures (since its bump that basically means turrets, and I guess hivecore). There needs to be a proper check for xeno structures in general.

Reply:
You just added another istype, instead of removing the existing one. I'm not going to hold you up on this since this definitely is an issue elsewhere as well, but this is still smelly and needs fixing in the future.